proto/spacemesh/global_state.proto

```diff
@@ -3,17 +3,48 @@ package spacemesh;
 import "google/protobuf/empty.proto";
 import "spacemesh/types.proto";
 
-// Readonly global state data.
+// Readonly global state data - current and historical.
 // Global state data is data which is not explicitly stored in the mesh.
 // Global state is modified only by the state transition function.
 service GlobalStateService {
-    //// Streaming global state data
+    // Latest computed global state - layer and its root hash
+    rpc GlobalState(google.protobuf.Empty) returns (GlobalState);
 
-    // Account changes (e.g., balance and counter/nonce changes).
-    rpc AccountStream(google.protobuf.Empty) returns (stream Account);
+    // Get the current transaction state. Note that tx state is not part of
+    // global state. It is metadata about a transaction on its journey to the
+    // mesh and from the mesh to the global state (or to null)
+    rpc TransactionState(TransactionId) returns (TransactionState);
+
+    // Account info in the current global state
+    rpc Account(AccountId) returns (Account);
+
+    // Rewards awarded to an account
+    rpc AccountRewards(AccountIdWithOffset) returns (Rewards);
+
+    // Transaction recipts for transactions to/from an account
+    rpc AccountTransactionReceipts(AccountIdWithOffset) returns (TransactionReceipts);
+
+    // Rewards awarded to a smesher
+    rpc SmesherRewards(SmesherIdWithOffset) returns (Rewards);
+
+    ////////// Streams
+
+    // Rewards awarded to a specific account
+    rpc AccountRewardStream(AccountId) returns (stream Reward);
+
+    // Rewards awarded to a smesher id
+    rpc SmesherRewardStream(SmesherId) returns (stream Reward);
+
+    // Returns a stream of Account changes. e.g. balance and counter
+    // (nonce) changes
+    rpc AccountStream(AccountId) returns (stream Account);
```

Comment:
This will create a stream that outputs a data item on each account balance change in the system and not just for accounts that the wallet is interested at. It will be very wasteful and the wallet will need to process each incoming message and check if it is for one of the addresses it is interested at - instead - the pattern is that it subscribes just on the accounts it is interested in knowing about and only get applicable messages that it cares about when they happen over the stream - much more efficient, less messages and less processing...

Reply:
Let's take this to #26